kratos/integration/integration_point_utilities.cpp

```diff
@@ -42,6 +56,44 @@ namespace Kratos
         }
     }
 
+    /* @brief Creates integration points within provided spans according
+     *        to the provided points per span.
+     * @param rIntegrationPoints result integration points.
+     * @param rSpansLocalSpace spans in which integration points are generated.
+     * @param IntegrationPointsPerSpan number of integration points per span.
+     */
+    void IntegrationPointUtilities::CreateIntegrationPoints1DGrid(
+        IntegrationPointsArrayType& rIntegrationPoints,
+        const std::vector<double>& rSpansLocalSpace,
+        const SizeType IntegrationPointsPerSpan)
+    {
+        const SizeType num_spans = rSpansLocalSpace.size() - 1;
+        const SizeType number_of_integration_points = num_spans * IntegrationPointsPerSpan + num_spans + 2;
+
+        if (rIntegrationPoints.size() != number_of_integration_points)
+            rIntegrationPoints.resize(number_of_integration_points);
+
+        IndexType counter = 0;
+        double last_increment = 0.0;
+        for (IndexType i = 0; i < num_spans; ++i)
+        {
+            const double increment = std::abs(rSpansLocalSpace[i + 1] - rSpansLocalSpace[i]) / IntegrationPointsPerSpan;
+            rIntegrationPoints[counter].X() = rSpansLocalSpace[i];
+            rIntegrationPoints[counter].Weight() = (increment / 2) + (last_increment / 2);
```

Comment:
Lets do moment fitting here :rocket: .

Reply:
hehe. Well, here I was about to focus on post-processing. Not sure if this will ease something then ;)

Nevertheless, we should extend explicitly those functions with your integration. And maybe add a check for moment fitting - to apply it once wanted?